### ISSUE TITLE
[Accessibility]: Unhelpful label for Sourcegraph logo

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#35146]
### Audit type

Screen reader navigation

### Problem description

The Sourcegraph logo on the top-left of Sourcegraph.com is read as "Sourcegraph logo" by VoiceOver. However, "logo" doesn't convey enough information to the user on where following the link will take them.

Additionally, it seems like VoiceOver sometimes messes up the pronunciation of Sourcegraph (noticed this on a call with @Strum355).

![Accessibility settings for Sourcegraph logo on top left](https://user-images.githubusercontent.com/93103176/167479074-bb42f6d9-cb91-4f34-ab1c-554ce6a0cef9.png)

### Expected behavior

Arguably, it would probably be more helpful for the label to be "Source graph home page" or similar.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
This is our team, we'll action it - Thanks!
